settings: crash from uninitialized ReportingSettingsSingleton

I saw this panic while attempting to call `log.Fatal` in `Server.Start` from a test in the `sql_test` package. I believe it's fallout from #17486.

```
panic: interface conversion: interface {} is nil, not *log.ReportingSettings [recovered]
	panic: interface conversion: interface {} is nil, not *log.ReportingSettings [recovered]
	panic: interface conversion: interface {} is nil, not *log.ReportingSettings

goroutine 20 [running]:
testing.tRunner.func1(0xc42022c750)
	/Users/nathan/Go/go/src/testing/testing.go:622 +0x29d
panic(0x54cc940, 0xc4207abcc0)
	/Users/nathan/Go/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:69 +0x4b3
panic(0x54cc940, 0xc4207abcc0)
	/Users/nathan/Go/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x82411f8, 0xc420132450, 0x4, 0x2, 0x0, 0x0, 0xc421021210, 0x1, 0x1)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:154 +0x552
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x82411f8, 0xc420132450, 0x1, 0x4, 0x0, 0x0, 0xc421021210, 0x1, 0x1)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:66 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x82411f8, 0xc420132450, 0xc421021210, 0x1, 0x1)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:169 +0x76
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0xc4200f1900, 0x82411f8, 0xc420132450, 0x7996cf8, 0xc420014e38)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1015 +0x268b
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc420478660, 0x6142320, 0xc4200a83f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:302 +0x184
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerRaw(0x6142320, 0xc4200a83f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:189 +0xef
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer(0x61655a0, 0xc42022c750, 0x6142320, 0xc4200a83f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:159 +0x55
github.com/cockroachdb/cockroach/pkg/sql_test.TestTxnAutoRetryParallelStmts(0xc42022c750)
	/Users/nathan/Go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:611 +0x10a
testing.tRunner(0xc42022c750, 0x5765428)
	/Users/nathan/Go/go/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/Users/nathan/Go/go/src/testing/testing.go:697 +0x2ca
```

cc @tschottdorf

--- a/pkg/migration/sqlmigrations/main_test.go
+++ b/pkg/migration/sqlmigrations/main_test.go
@@ -21,20 +21,12 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/security"
 	"github.com/cockroachdb/cockroach/pkg/security/securitytest"
 	"github.com/cockroachdb/cockroach/pkg/server"
-	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/util/log"
 )
 
 func TestMain(m *testing.M) {
 	security.SetAssetLoader(securitytest.EmbeddedAssets)
 	serverutils.InitTestServerFactory(server.TestServerFactory)
-
-	// TestLeaseExpiration needs log.ReportingSettingsSingleton populated as it
-	// calls log.Fatal.
-	st := cluster.MakeClusterSettings()
-	f := log.ReportingSettings(st.ReportingSettings)
-	log.ReportingSettingsSingleton.Store(&f)
 
 	os.Exit(m.Run())
 }

--- a/pkg/util/log/structured.go
+++ b/pkg/util/log/structured.go
@@ -151,8 +151,9 @@ func addStructured(ctx context.Context, s Severity, depth int, format string, ar
 
 		// We load the ReportingSettings from the a global singleton in this
 		// call path. See the singleton's comment for a rationale.
-		reportingSettings := (ReportingSettingsSingleton.Load()).(*ReportingSettings)
-		sendCrashReport(ctx, *reportingSettings, reportable, depth+1)
+		if reportingSettings, ok := (ReportingSettingsSingleton.Load()).(*ReportingSettings); ok {
+			sendCrashReport(ctx, *reportingSettings, reportable, depth+1)
+		}
 	}
 	// MakeMessage already added the tags when forming msg, we don't want
 	// eventInternal to prepend them again.

